lebedigital/openbis/expstep.py

```diff
@@ -0,0 +1,1464 @@
+                for key, val in value.items():
+                    colkey = colored(key, 'green', 'on_grey')
+                    print(' '*20 + f'{colkey}: {val}')
+                print('\n')
+            else:
+                print(f'{colparam: >25}: {value}' + '\n')
+
+        print('=' * int(float(len(title)) * 0.9))
+
+    @staticmethod
+    def connect_to_datastore(url='https://test.datastore.bam.de/openbis/', *args, **kwargs):
+        """Connects to a Datastore
+
+        Args:
+            url (str, optional): Datastore URL.
+                    Defaults to 'https://test.datastore.bam.de/openbis/'.
+
+        Returns:
+            Openbis: Connected and running Datastore instance
+        """
+        os.environ['OPENBIS_HOST'] = url
+        o = Openbis(os.environ['OPENBIS_HOST'])
+
+        # If a session is already active (when running a notebook again) return the active object
+        if o.is_session_active():
+            return o
+
+        # We get the username from the os username and the password has to be given by the user at runtime
+        else:
+
+            if 'username' in kwargs:
+                os.environ['OPENBIS_USERNAME'] = kwargs['username']
+            else:
+                os.environ['OPENBIS_USERNAME'] = os.getlogin()
+
+            if 'password' in kwargs:
+                os.environ['OPENBIS_PASSWORD'] = kwargs['password']
+            else:
+                os.environ['OPENBIS_PASSWORD'] = getpass("Give Password: ")
+
+            try:
+                o.login(os.environ['OPENBIS_USERNAME'],
+                        os.environ['OPENBIS_PASSWORD'])
+            except ValueError:
+                print("Wrong Credentials")
+                exit(1)
+
+            return o
+
+    @staticmethod
+    def read_metadata_json(json_path: str) -> dict:
+        """Read the metadata from a json file. You can download the metadata as json from the datastore website.
+
+        Args:
+            json_path (str): Path to json file
+
+        Returns:
+            dict: Json file as dict. Can be directly imported into the
+                  datastore.
+        """
+        with open(json_path, 'r') as f:
+            metadata = json.load(f).get('properties')
+        # making keys lowercase, k - key, v - value
+        metadata = {k.lower(): v for k, v in metadata.items()}
+        return metadata
+
+    @staticmethod
+    def gen_metadata_import_template(
+        o: Openbis,
+        sample_type: str,
+        write: bool = False,
+        sheet_name: str = 'metadata',
+        path: str = '',
+    ):
+        """ Generate an excel data import template for a given object type.
+
+        Args:
+            o (Openbis): Currently running openbis instance
+            sample_type (str): The type of the sample (ex. EXPERIMENTAL_STEP_SOMETHING)
+            write (bool, optional): Specifies if the function should return the template as a pandas DataFrame or if it should be written to an excel file. Defaults to False.
```

Comment:
can you format your file (e.g. using [black](https://github.com/psf/black) with style [black](https://black.readthedocs.io/en/stable/the_black_code_style/current_style.html) (e.g. the lines should not be longer than e.g. 88 characters)